### PR TITLE
Add support for esp32c3 hardware half duplex RS485 mode

### PR DESCRIPTION
**Pull request by dean-at-evos**
Hardware flow control mode UART_CFG_FLOW_CTRL_RS485 is enabled for the esp32 series. Option hw-rs485-hd-mode for the UART now enables DTR pin toggling during transmit. Also fixes bug where flow control modes weren't applied at boot.

Signed-off-by: Dean Sellers <dsellers@evos.com.au>

### CODE REVIEW

**Comment by dean-at-evos** [comment]
Updated PR to include all ESP32 SOC.

**Comment by sylvioalves** [comment]
@dean-at-evos Will you fix this PR?

**Comment by dean-at-evos** [comment]
Yes, sorry. Am a bit new to this. I noticed it had failed due to the HAL changes necessary, but now due to my bad formatting. Will try and address today.

**Comment by dean-at-evos** [comment]
@sylvioalves Thanks for prompting me on this one, this should be OK now. Had to brush up on my git-fu :)

File: west.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: d2564ca5c14439888f7c4351884a2cf4db517a33
+      revision: 029021f8a6f4b423c5c1acfe961e8c988da5e3eb
```
[the remark this answers is not in the document]
Comment (dean-at-evos) [reply west.yml]: Done

**Review by sylvioalves — DISMISSED** [review]

**Comment by marekmatej** [comment]
@dean-at-evos almost there, can you squeeze last two commits with manifest change into single commit please?

**Review by sylvioalves — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**